src/Admin/Tools/SendMailToUser/SendMailToUserController.php

```diff
@@ -51,4 +53,50 @@ public function sendAction(Request $request): Response
 
     return new Response('OK - message sent');
   }
+
+  public function previewAction(Request $request): Response
+  {
+    $resetToken = $this->resetPasswordHelper->generateFakeResetToken();
+
+    $signature = 'https:://example.url';
+
+    $template = (string) $request->query->get('template');
+
+    $user = $this->user_manager->findUserByUsername((string) $request->query->get('username'));
+    if (!$user) {
+      return new Response('User does not exist');
+    }
+
+    $subject = (string) $request->query->get('subject');
+    if ('' === $subject && '2' === $template) {
+      return new Response('Empty subject!');
+    }
+
+    $titel = (string) $request->query->get('titel');
+
+    $messageText = (string) $request->query->get('message');
+    if ('' === $messageText && '2' === $template) {
+      return new Response('Empty message!');
+    }
```

Comment:
would be nice to execute each snippet only if it is necessary - think of this previewAction as delegator 

switch $template:
   case A: load fake data for A - render
   case B: load fake data for B - render
   
 Feel free to create multiple classes or at least methods in a way that each class/method only has one job